src/classes/CON_Preferred_TDTM.cls

```diff
@@ -36,42 +36,36 @@
 */
 public class CON_Preferred_TDTM extends TDTM_Runnable {
 
-	/*******************************************************************************************************
+    /*******************************************************************************************************
     * @description Updates default email and phone fields based to the preferred email and preferred phone values.
     * @return dmlWrapper
     ********************************************************************************************************/
-	public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
         TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
-        	 
-    	 if((triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) && newlist != null) {
-    	 	
-    	 	for(SObject so : newlist) {
-    	 		Contact contact = (Contact)so;
-    	 	
-	    	 	if(!String.isBlank(contact.Preferred_Email__c)) {
-	    	 		if(contact.Preferred_Email__c == 'University') {
-	    	 			contact.Email = contact.UniversityEmail__c;
-	    	 		} else if(contact.Preferred_Email__c == 'Work') {
-	    	 			contact.Email = contact.WorkEmail__c;
-	    	 		} else if(contact.Preferred_Email__c == 'Alternate') {
-	    	 			contact.Email = contact.AlternateEmail__c;
-	    	 		}
-	    	 	}
-	    	 	
-	    	 	if(!String.isBlank(contact.PreferredPhone__c)) {
-	    	 		if(contact.PreferredPhone__c == 'Home') {
-	    	 			contact.Phone = contact.HomePhone;
-	    	 		} else if(contact.PreferredPhone__c == 'Work') {
-	    	 			contact.Phone = contact.WorkPhone__c;
-	    	 		} else if(contact.PreferredPhone__c == 'Mobile') {
-	    	 			contact.Phone = contact.MobilePhone;
-	    	 		} else if(contact.PreferredPhone__c == 'Other') {
-	    	 			contact.Phone = contact.OtherPhone;
-	    	 		}
-	    	 	}
-    	 	}
-    	 }
-    	 
-    	 return null;
-   	}
+             
+        if((triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) && newlist != null) {
+
+            for(SObject so : newlist) {
+                Contact contact = (Contact)so;
+
+                // Process the prefered Email
+                CON_Email.processPreferredEmail(contact, oldlist);
+                
+                // Process the Preferred Phone
+                if(!String.isBlank(contact.PreferredPhone__c)) {
```

Comment:
I think you just made whitespace changes in this section, but we should open a work item to make this translatable. It has the same problem mentioned above, where the dropdown could be modified and this would stop working. Maybe we can tackle both issues at the same time.